drivers/adc/ad400x/iio_ad400x.c

```diff
@@ -0,0 +1,237 @@
+		return ret;
+
+	if (scan_type->sign == 's') {
+		results_signed = no_os_sign_extend32(results, scan_type->realbits - 1);
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &results_signed);
+	}
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, (int32_t *)&results);
+}
+
+static int get_scale(void *device, char *buf, uint32_t len,
+		     const struct iio_ch_info *channel, intptr_t id)
+{
+	struct ad400x_iio_dev *iio_dev = device;
+	struct scan_type *scan_type;
+	int32_t vals[2];
+
+	scan_type = iio_dev->iio_dev->channels[channel->ch_num].scan_type;
+	vals[0] = iio_dev->ref_voltage_mv;
+	if (scan_type->sign == 'u')
+		vals[1] = scan_type->realbits;
+	else
+		vals[1] = scan_type->realbits - 1;
+
+	return iio_format_value(buf, len, IIO_VAL_FRACTIONAL_LOG2, 2, vals);
+}
+
+/**
+ * @brief Read buffer data corresponding to AD400X IIO device
+ * @param iio_dev_data - Pointer to IIO device data structure
+ * @return 0 in case of success, negative error code otherwise
+ */
+static int32_t iio_ad400x_submit_buffer(struct iio_device_data *iio_dev_data)
+{
+	struct ad400x_iio_dev *iio_dev = iio_dev_data->dev;
+	struct ad400x_dev *dev = iio_dev->ad400x_dev;
+	struct iio_buffer *buffer = iio_dev_data->buffer;
+	void *buff;
+	int ret;
+
+	ret = iio_buffer_get_block(iio_dev_data->buffer, &buff);
+	if (ret)
+		return ret;
+
+	ret = ad400x_read_data(dev, buff, buffer->samples);
+	if (ret)
+		return ret;
+
+	return iio_buffer_block_done(buffer);
+}
+
+static struct iio_attribute ad400x_iio_ch_attributes[] = {
+	{ .name = "raw", .show = get_raw },
+	{ .name = "scale", .show = get_scale, },
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_channel ad400x_channel = {
+	.name = "voltage0",
+	.ch_type = IIO_VOLTAGE,
+	.ch_out = IIO_DIRECTION_INPUT,
+	.indexed = true,
+	.channel = 0,
+	.scan_index = 0,
+	.attributes = ad400x_iio_ch_attributes,
+};
+
+static struct iio_device ad400x_iio_device_template = {
+	.num_ch = 1,
+	.channels = &ad400x_channel,
+	.debug_reg_read = iio_ad400x_debug_reg_read,
+	.debug_reg_write= iio_ad400x_debug_reg_write,
+	.submit = iio_ad400x_submit_buffer,
+};
+
+/**
+ * @brief Initialize AD400X for IIO interfacing
+ * @param dev - The device structure.
+ * @param iio_init_param - IIO init parameter structure
+ * @return 0 on success, an error code otherwise
+ */
+int ad400x_iio_init(struct ad400x_iio_dev **dev,
+		    struct ad400x_iio_init_param *iio_init_param)
+{
+	struct ad400x_iio_dev *desc;
+	enum ad400x_supported_dev_ids dev_id;
+	int ret;
+
+	desc = no_os_calloc(1, sizeof(*desc));
+	if (!desc)
+		return -ENOMEM;
+
+	desc->iio_dev = &ad400x_iio_device_template;
+	desc->ref_voltage_mv = iio_init_param->ref_voltage_mv;
+
+	dev_id = iio_init_param->init_param->dev_id;
+
+	/* fill scan_type based on device id */
+	desc->scan_type.sign = ad400x_device_sign[dev_id];
+	desc->scan_type.realbits = ad400x_device_resol[dev_id];
+	desc->scan_type.storagebits = 32;
+	desc->scan_type.is_big_endian = false;
+	desc->scan_type.shift = 0;
+
+
```

Comment:
remove extra lines

Reply:
this extra line is not part of the latest push.

Reply:
it is part of this commit: https://github.com/analogdevicesinc/no-OS/pull/2105/commits/5e09916ac4bbffcf776f241d2fdb8684b814824e 
in general, in the same PR, we avoid fixing a problem we introduced with a later commit, rather we remove it in the commit it was introduced by rebasing.
this makes the review process easier, commit by commit

Reply:
ok,  ill fix it there.